src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteTagHelperResolver.cs

```diff
@@ -4,53 +4,53 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.AspNetCore.Razor.Serialization;
 using Microsoft.AspNetCore.Razor.Telemetry;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
 
 namespace Microsoft.CodeAnalysis.Razor;
 
-internal class RemoteTagHelperResolver : TagHelperResolver
+internal class RemoteTagHelperResolver : ITagHelperResolver
 {
     private readonly static RazorConfiguration s_defaultConfiguration = FallbackRazorConfiguration.MVC_2_0;
 
     private readonly IFallbackProjectEngineFactory _fallbackFactory;
+    private readonly CompilationTagHelperResolver _compilationTagHelperResolver;
 
     public RemoteTagHelperResolver(IFallbackProjectEngineFactory fallbackFactory, ITelemetryReporter telemetryReporter)
-        : base(telemetryReporter)
     {
-        if (fallbackFactory is null)
-        {
-            throw new ArgumentNullException(nameof(fallbackFactory));
-        }
-
-        _fallbackFactory = fallbackFactory;
+        _fallbackFactory = fallbackFactory ?? throw new ArgumentNullException(nameof(fallbackFactory));
```

Comment:
I still haven't figured out the pattern for using primary constructors or not. I'm feeling like a curmudgeonly coder 🧓

Reply:
IMO the only valid pattern is using them to initialize members, or pass to base constructors. Using them in methods should be banned 😁

Reply:
> I'm feeling like a curmudgeonly coder

I know the feeling! :smile: